### PR TITLE
fix: displays the blocks of the Section Metadata page

### PR DESCRIPTION
**Pull request by jckautzmann**
- displays the blocks of the Section Metadata page

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/tools/library?martech=off
- After: https://library-search-fix--milo--adobecom.hlx.page/tools/library?martech=off

### CODE REVIEW

**Comment by jckautzmann** [comment]
The failing test seems unrelated to the PR changes:
```
test/blocks/marketo/marketo.test.js:

 ❌ marketo > load marketo disables stylesheets
      Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
        at createTimeoutError (node_modules/@web/test-runner-mocha/dist/autorun.js:1:243195)
        at Gg._timeoutError (node_modules/@web/test-runner-mocha/dist/autorun.js:1:259145)
        at node_modules/@web/test-runner-mocha/dist/autorun.js:1:257217
```

**Review by ryanmparrish — APPROVED** [review]

**Comment by auniverseaway** [comment]
@rgclayton @jckautzmann are we using section-metadata for the tagging / search feature?

This would mean that section-metadata is going to show on every single block page... which is not ideal, and this PR is trying to address.

If we look long term: block library should really be more like, "pattern" library where you can bring in groups of blocks in a single click. Think of tabs or carousel right now. They are not supported in library because they are composed of many blocks.

I would walk back from that:

We want section-metadata to show up in the block library, but only when we want it to.

Should we make a "faux" block that's called "library-metadata" that we _could_ opt out of cleanly and then section-metadata can keep being available when we want? I made `getMetadata` relatively extensible so we could do this.

File: libs/blocks/library-config/lists/blocks.js [hunk 1]
```diff
@@ -101,7 +101,9 @@ export default async function loadBlocks(blocks, list, query) {
     const pageBlocks = doc.body.querySelectorAll('div[class]');
     let matchingBlockFound = false;
     pageBlocks.forEach((pageBlock) => {
-      if (pageBlock.className === 'section-metadata') {
+      // don't display the section-metadata used to set the block search tags
+      // unless the block is part of the Section Metadata page
+      if (block.name !== 'Section Metadata' && pageBlock.className === 'section-metadata') {
```
[the remark this answers is not in the document]
Comment (jckautzmann) [reply libs/blocks/library-config/lists/blocks.js:106]: @rgclayton Do you have examples of blocks for which we display their `section-metadata` in the library, apart from the blocks in the `Section Metadata` page? If there are none I would keep the condition as it is until we enhance the library so one can add a group of blocks as @auniverseaway mentioned below. I would only introduce the `library-metadata` when we have a real need for it. WDYT?

[the remark this answers is not in the document]
Comment (rgclayton) [reply libs/blocks/library-config/lists/blocks.js:106]: Yeah, we have two. Tabs and Carousel blocks.

I think I'm up to speed now after reading Chris' comment :). 
I think this PR is on the right track. What I think we will do after this is create a new `library-metadata` block, like Chris mentioned, to cleanly separate search from `section-metadata`.  We would then add the search tags to `library-metadata` and then be able to use this for all things specific to the library.

**Comment by amauch-adobe** [comment]
> The failing test seems unrelated to the PR changes:
> 
> ```
> test/blocks/marketo/marketo.test.js:
> 
>  ❌ marketo > load marketo disables stylesheets
>       Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.
>         at createTimeoutError (node_modules/@web/test-runner-mocha/dist/autorun.js:1:243195)
>         at Gg._timeoutError (node_modules/@web/test-runner-mocha/dist/autorun.js:1:259145)
>         at node_modules/@web/test-runner-mocha/dist/autorun.js:1:257217
> ```

@jckautzmann after updating the PR to pull in the latest main branch updates all of the unit tests have passed.  I think the timeout was a false failure where something just failed to correctly respond back in the test's callbacks.  I believe we are good here, plus as you mentioned this unit test isn't even reflected in the changes you have made or neither in the marketo block's code itself.

**Review by rgclayton — APPROVED** [review]

**Comment by jckautzmann** [comment]
@amauch-adobe Thx for having looked into the failing test to make sure it's not related to this PR.

**Comment by chrischrischris** [comment]
Do we merge this now or use this to create "library-metadata"?
@auniverseaway @jckautzmann @rgclayton

**Comment by rgclayton** [comment]
> Do we merge this now or use this to create "library-metadata"? @auniverseaway @jckautzmann @rgclayton

I was just thinking about this. We do have someone from Rikio's team putting together searchable ID's to add to blocks `section-metadata`. If we add the `library-metadata` later that means that work will have to be updated for all blocks. 

I'm inclined to use this to create `library-metadata` to avoid any duplicate work.

Any strong opinions @auniverseaway @jckautzmann @chrischrischris?

**Comment by chrischrischris** [comment]
> I'm inclined to use this to create library-metadata to avoid any duplicate work.
Agreed @rgclayton

**Comment by rgclayton** [comment]
> I'm inclined to use this to create library-metadata to avoid any duplicate work.

> Agreed @rgclayton

@jckautzmann Sound ok? Any questions/concerns?

**Comment by jckautzmann** [comment]
> I'm inclined to use this to create library-metadata to avoid any duplicate work.

@rgclayton +1, I'll work on it tomorrow.

**Comment by jckautzmann** [comment]
@rgclayton As discussed I introduced a library-metadata block in https://github.com/adobecom/milo/pull/472/commits/e2245e7b3890e9234891818a3436e4be16c9d5bf

**Comment by rgclayton** [comment]
@jckautzmann Looking good!

A couple things. 

1. We need to take care of the "Failed loading library-metadata block"
 image
2. Double checking that we can have the `library-metadata` block per each block variation? Authors will most likely want to add special tags per variation. This would be similar to `section-metadata` block being coupled with each variation for unique tags.

cc: @auniverseaway

**Review by auniverseaway — APPROVED** [review]

**Pull request merged**